src/whylogs/whylabs_client/wrapper.py

```diff
@@ -37,30 +57,56 @@ def start_session() -> None:
 
 
 def upload_profile(dataset_profile: DatasetProfile) -> None:
-    dataset_timestamp = None
+    client = _get_whylabs_client()
+
+    tmp_dir = tempfile.mkdtemp()
+    profile_path = os.path.join(tmp_dir, "profile.bin")
+    dataset_profile.write_protobuf(profile_path)
+
+    dataset_timestamp = dataset_profile.dataset_timestamp or datetime.datetime.now(datetime.timezone.utc)
+    dataset_timestamp = int(dataset_timestamp.timestamp() * 1000)
+    if _session_token:
+        _upload_guest_session(client, dataset_timestamp, profile_path)
+    else:
+        _upload_whylabs(dataset_profile, dataset_timestamp, profile_path)
+
+
+def _upload_whylabs(dataset_profile, dataset_timestamp, profile_path):
+    _logger.debug("Upload with WhyLabs API token")
+    log_api = _get_or_create_log_client()
+    org_id = dataset_profile.tags.get("orgId", os.environ.get("WHYLABS_DEFAULT_ORG_ID"))
+    dataset_id = dataset_profile.tags.get("datasetId", os.environ.get("WHYLABS_DEFAULT_DATASET_ID"))
+    if org_id is None:
+        raise EnvironmentError('Missing organization ID. Specify it via "orgId" tag or WHYLABS_DEFAULT_ORG_ID environment variable')
+    if dataset_id is None:
+        raise EnvironmentError('Missing dataset ID. Specify it via "datasetId" tag or WHYLABS_DEFAULT_DATASET_ID environment variable')
```

Comment:
should the tags also be in snake case ?

Reply:
No. This is for the backend and we use camelCase everywhere else